snapshot/git/gitdb/backends.go

```diff
@@ -0,0 +1,62 @@
+package gitdb
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/scootdev/scoot/snapshot"
+)
+
+type backend interface {
+	parseID(id string, kind string, parts []string) (snap, error)
+}
+
+type uploader interface {
+	upload(id snapshot.ID) (snapshot.ID, error)
+	backend
+}
+
+// GitDB uses several different Backends to store Snapshots.
+// snap is the common interface for a Snapshot (identified by an ID)
+type snap interface {
```

Comment:
snapID?

I'm happy to delete snapshot/interface.go for now, along with the FS code.... But maybe that's a separate PR?

This isn't just a generic snapshot, this is the implementation of snapshot in GitDB, so we could call it gitDBSnap, but it's already in package gitdb.